packages/vsce/src/SshClientCache.ts

```diff
@@ -38,22 +46,43 @@ export class SshClientCache extends vscode.Disposable {
         if (restart) {
             this.end(clientId);
         }
+
         if (!this.mClientMap.has(clientId)) {
             const session = ZSshUtils.buildSession(profile.profile!);
             const serverPath = SshConfigUtils.getServerPath(profile.profile);
-            this.mClientMap.set(
-                clientId,
-                await ZSshClient.create(session, {
+            const localDir = path.join(this.mContext.extensionPath, "bin");
+            const autoUpdate = vscode.workspace
+                .getConfiguration("zowe-native-proto-vsce")
+                .get("serverAutoUpdate", true);
+
+            let newClient: ZSshClient;
+            try {
+                newClient = await this.buildClient(session, clientId, {
                     serverPath,
-                    onClose: () => {
-                        this.end(clientId);
+                    onConnect: async (client: Client) => {
+                        if (
+                            autoUpdate &&
+                            (await ZSshUtils.checkIfOutdated(client, serverPath, path.join(localDir, "checksums.asc")))
+                        ) {
+                            imperative.Logger.getAppLogger().info(
+                                `Server is out of date, deploying to ${profile.name}`,
+                            );
+                            await deployWithProgress(session, serverPath, localDir);
+                        }
```

Comment:
could we have an `else` to at least warn the user about the server being out of date (in case they turn off the setting) 😋